src/main/java/daos/DBPostDataAccessObject.java

```diff
@@ -0,0 +1,199 @@
+            Updates.set(CONTENT_BODY, updatedContent.getContent().getBody()),
+            Updates.set(ATTACHMENT_PATH, updatedContent.getContent().getAttachmentPath()),
+            Updates.set(FILE_TYPE, updatedContent.getContent().getFileType()),
+            Updates.set(CATEGORY, updatedContent.getCategory()),
+            Updates.set(LAST_MODIFIED, updatedContent.getLastModifiedDate()),
+            Updates.set(LIKES, updatedContent.getLikes()),
+            Updates.set(DISLIKES, updatedContent.getDislikes()),
+            Updates.set(COMMENTS, updatedContent.getComments()) // TODO type conversion? need testing
+        );
+
+        // Instructs the driver to insert a new document if none match the query
+        UpdateOptions insertNewDoc = new UpdateOptions().upsert(true);
+
+        try {
+            UpdateResult result = this.postRepository.updateOne(query, updates, insertNewDoc);
+            System.out.println("Upserted id: " + result.getUpsertedId());
+        } catch (MongoException error) {
+            // throw err?
+        }
+    }
+
+    
+    /**
+     * Inserts the given post into the database.
+     * @param user - a user in the application.
+     */
+    private void insertPostToDB(Post post) {
+        try {
+            
+            Document data = (new Document()
+                .append(ENTRY_ID, post.getEntryID())
+                .append(AUTHOR, post.getAuthor())
+                .append(CONTENT_BODY, post.getContent().getBody())
+                .append(ATTACHMENT_PATH, post.getContent().getAttachmentPath())
+                .append(FILE_TYPE, post.getContent().getFileType())
+                .append(POST_TITLE, post.getPostTitle())
+                .append(CATEGORY, post.getCategory())
+                .append(POSTED_DATE, post.getPostedDate())
+                .append(LAST_MODIFIED, post.getLastModifiedDate())
+                .append(LIKES, post.getLikes())
+                .append(DISLIKES, post.getDislikes())
+                .append(COMMENTS, post.getComments()) // TODO figure out type conversions if neccessary
+            );
+
+            InsertOneResult result = this.postRepository.insertOne(data);
+            System.out.println("Successfully inserted post with insert id: " + result.getInsertedId());
+        } catch (MongoException err) {
+            // TODO throws custom exceptions when they're created
+        }
+    }
+
+    /**
+     * Queries a specific post from the database.
+     * @param field - the column to to match.
+     * @param target - the target value to query for.
+     */
+    private Document queryOnePostBy(String field, String target) {
+        Bson projectionFields = Projections.fields(
+            Projections.include(field),
+            Projections.excludeId()
+        );
+        
+        Document doc = this.postRepository
+            .find(eq(field, target))
+            .projection(projectionFields)
+            .first();
+
+        return doc;
```

Comment:
So this code returns a Document that only contains the field `field` right?

E.g. if we searched for a post with entryID 1, the document returned would be {'entryID': 1}? Is this what is expected or do we want to return the entire Post itself (so all fields?)

Reply:
So line 169 does not only include the field that was inputted?

Reply:
no, the field is the thing we search and match for in the database

Reply:
Ok!